app/controllers/declaration/SealController.scala

```diff
@@ -55,13 +55,18 @@ class SealController @Inject()(
   }
 
   def submitAddSeal(mode: Mode, containerId: String): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
-    val boundForm = Seal.form().bindFromRequest()
-
-    request.cacheModel.containerBy(containerId) match {
-      case Some(container) =>
-        saveSeal(mode, boundForm, container)
-      case _ => errorHandler.displayErrorPage()
-    }
+    Seal
+      .form()
+      .bindFromRequest()
+      .fold(
+        (formWithErrors: Form[Seal]) => Future.successful(BadRequest(addPage(mode, formWithErrors, containerId))),
```

Comment:
We have this part in `saveSeal` method. It probably means that we don't need to check once again errors in `saveSeal` method

Reply:
I think the form errors in saveSeal will be a result of calling saveAndContinue - i.e. list max length validation errors.    So I think both are needed